test/bond-factory.test.ts

```diff
@@ -44,6 +52,18 @@ describe('Bond Factory contract', () => {
         )
     })
 
+    describe.only('proxy upgrade', () => {
+        it('maintains version across upgrade', async () => {
+            const originalVersion = await bonds.VERSION()
+            await upgradeContract('BondFactory', bonds.address)
+            const upgradedVersion = await bonds.VERSION()
+            expect(originalVersion).to.equal(upgradedVersion)
+            // dummyUpgradedBondFactory = await deployContract<UpgradedVersion>("UpgradedVersion")
+            await upgradeContract('UpgradedVersion', bonds.address)
```

Comment:
this gives:
`       proxy upgrade
         maintains version across upgrade:
     Error: Requested an upgrade of kind transparent but proxy is uups
      at setProxyKind (node_modules/@openzeppelin/upgrades-core/src/proxy-kind.ts:26:11)
      at async processProxyKind (node_modules/@openzeppelin/upgrades-core/src/proxy-kind.ts:53:5)
      at async deployProxyImpl (node_modules/@openzeppelin/hardhat-upgrades/src/utils/deploy-impl.ts:68:3)
      at async Proxy.upgradeProxy (node_modules/@openzeppelin/hardhat-upgrades/src/upgrade-proxy.ts:25:32)
      at async upgradeContract (test/framework/contracts.ts:56:35)
      at async Context.<anonymous> (test/bond-factory.test.ts:62:13)`

Reply:
oz gives (defaults?) this error when the contract you are attempting to upgrade to does not inherit any upgradeability features?